cmd/sql/flow.go

```diff
@@ -8,279 +8,255 @@ import (
+// Initialize persistent/global flags inserted before the cmd
+func initGlobalCmdArgs() []string {
+	var args []string
+	if debug {
+		args = append(args, "--debug")
 	}
-
-	err = os.MkdirAll(projectDir, os.ModePerm)
-
-	if err != nil {
-		err = fmt.Errorf("error creating project directory %s: %w", projectDir, err)
-		return "", err
+	if noDebug {
+		args = append(args, "--no-debug")
 	}
+	return args
+}
 
-	return projectDir, nil
+// Initialize specific cmd args by setting the cmd flags, resolving filepaths and overwriting args
+func initLocalCmdArgs(cmd *cobra.Command, args []string) []string {
+	switch cmd.Name() {
+	case initCmdName:
+		projectDir = resolvePath(getProjectDirFromArgs(args))
+		if airflowHome != "" {
+			airflowHome = resolvePath(airflowHome)
+		}
+		if airflowDagsFolder != "" {
+			airflowDagsFolder = resolvePath(airflowDagsFolder)
+		}
+		if dataDir != "" {
+			dataDir = resolvePath(dataDir)
+		}
+		return []string{projectDir}
+	case configCmdName:
+		projectDir = resolvePath(projectDir)
+	case validateCmdName:
+		projectDir = resolvePath(getProjectDirFromArgs(args))
+		return []string{projectDir}
+	case generateCmdName:
+		projectDir = resolvePath(projectDir)
+	case runCmdName:
+		projectDir = resolvePath(projectDir)
+	}
+	return args
 }
 
-func getBaseMountDirs(projectDir string) ([]string, error) {
-	mountDir, err := createProjectDir(projectDir)
-	if err != nil {
-		return nil, err
+// Get the projectDir flag from args if given
+func getProjectDirFromArgs(args []string) string {
+	if len(args) > 0 {
+		return args[0]
 	}
-	mountDirs := []string{mountDir}
-	return mountDirs, nil
+	return "."
 }
 
-var appendConfigKeyMountDir = func(configKey string, configFlags map[string]string, mountDirs []string) ([]string, error) {
-	args := []string{configKey}
-	exitCode, output, err := sql.ExecuteCmdInDocker(configCommandString, args, configFlags, mountDirs, true)
-	if err != nil {
-		return mountDirs, fmt.Errorf("error running %v: %w", configCommandString, err)
-	}
-	if exitCode != 0 {
-		return mountDirs, sql.DockerNonZeroExitCodeError(exitCode)
-	}
-	configKeyDir, err := sql.ConvertReadCloserToString(output)
-	if err != nil {
-		return mountDirs, err
-	}
-	mountDirs = append(mountDirs, strings.TrimSpace(configKeyDir))
-	return mountDirs, nil
+// Read config cmd output for retrieving config settings such as airflow_home
+func readConfigCmdOutput(key string) string {
+	args := []string{key}
+	output := readCmdOutput(configCmd, args)
+	return strings.TrimSpace(output) // remove spaces such as \r\n
 }
 
-func buildFlagsAndMountDirs(projectDir string, setProjectDir, setAirflowHome, setAirflowDagsFolder, setDataDir, mountGlobalDirs bool) (flags map[string]string, mountDirs []string, err error) {
-	flags = make(map[string]string)
-	mountDirs, err = getBaseMountDirs(projectDir)
+// Resolve filepath to absolute
+func resolvePath(path string) string {
+	path, err := filepath.Abs(path)
 	if err != nil {
-		return nil, nil, err
+		panic(err)
```

Comment:
we should try to return the error instead of panic, always ideal to pass down the error instead of panicking, make the code much easier to maintain/read